### PR TITLE
PLUGINS - Merge fixes

### PR DESCRIPTION
**Pull request by jherico**
My latest merge from master left me crashing on startup.  I believe this resolves the problems with the GL initialization order.

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Build console output for commit: ccc4a599929b04dad560b823ed7950d37b1a3bc1 available at: https://gist.github.com/fe6afd0dbf3ebd903933<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5534.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5534.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: ba498e8f33dcdcf665d9a6a928a3eff8ddf0d090 available at: https://gist.github.com/fe6afd0dbf3ebd903933<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5534.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5534.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 639a0822fed1dfec16a52fe8192b86ef5288a8e8 available at: https://gist.github.com/fe6afd0dbf3ebd903933<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5534.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5534.dmg

File: libraries/shared/src/GLMHelpers.cpp [hunk 1]
```diff
@@ -313,27 +313,31 @@ bool isSimilarPosition(const glm::vec3& positionA, const glm::vec3& positionB, f
...
-glm::vec4 toGlm(const QColor & color) {
+glm::vec4 toGlm(const QColor& color) {
     return glm::vec4(color.redF(), color.greenF(), color.blueF(), color.alphaF());
 }
 
+ivec4 toGlm(const QRect& rect) {
+    return ivec4(rect.x(), rect.y(), rect.width(), rect.height());
+}
+
 QMatrix4x4 fromGlm(const glm::mat4 & m) {
```
Comment (ZappoMan) [inline libraries/shared/src/GLMHelpers.cpp]: this should have a type squish too... not a merge blocker

File: libraries/shared/src/GLMHelpers.h [hunk 2]
```diff
@@ -115,11 +118,12 @@ bool isSimilarOrientation(const glm::quat& orientionA, const glm::quat& orientio
...
-glm::vec3 toGlm(const xColor & color);
-glm::vec4 toGlm(const QColor & color);
+uvec2 toGlm(const QSize& size);
+ivec2 toGlm(const QPoint& pt);
+vec2 toGlm(const QPointF& pt);
+vec3 toGlm(const xColor& color);
+vec4 toGlm(const QColor& color);
+ivec4 toGlm(const QRect& rect);
 
 QSize fromGlm(const glm::ivec2 & v);
```
Comment (ZappoMan) [inline libraries/shared/src/GLMHelpers.h]: type squishes here too - not a merge blocker

**Comment by hifi-gustavo** [comment]
Build console output for commit: 751db23bc8a423ab35ca3ed5a746053c1747f1e0 available at: https://gist.github.com/fe6afd0dbf3ebd903933<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5534.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5534.dmg

**Pull request merged**